examples/threading/threading.c

```diff
@@ -1,33 +1,148 @@
+/****************************************************************************************************
+  File name   : threading.c
+
+  Created     : 15-Sep-2022
+
+  Author      : Kevin Tom
+                MS-ECEE-ESE
+                CU Boulder, CO
+
+  Email       : keto9919@colorado.edu
+
+  Description : Creates thread as requested and locks on a mutex till time expires.
+
+
+  Reference   : * Linux System Programming (Second Edition)
+                  Robert Love
+                * usleep() : https://linux.die.net/man/3/usleep
+******************************************************************************************************/
 #include "threading.h"
-#include <unistd.h>
+#include <unistd.h> 
 #include <stdlib.h>
 #include <stdio.h>
+#include <errno.h> //For error handling
+#include <time.h>  //Used for sleep() functions
+
 
 // Optional: use these functions to add debug or error prints to your application
 #define DEBUG_LOG(msg,...)
 //#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
 #define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
 
+
+#define CONVERT_SEC_TO_NS 1000*1000
+#define CONVERT_MS_TO_NS 1000
+
 void* threadfunc(void* thread_param)
 {
+    struct thread_data * thread_args = (struct thread_data *) thread_param;
+    struct timespec wait_rem; //Structure for remaining time
+    
+
+    nanosleep(&(thread_args->wait_to_obtain_ms),  &wait_rem);
+    
+    if(pthread_mutex_lock(thread_args->thread_mutex) != 0)
+    {
+        perror("pthread_mutex_lock");
+        thread_args->thread_complete_success = false;
+        return thread_param;
+    }
+
+    nanosleep(&(thread_args->wait_to_release_ms),  &wait_rem);
```

Comment:
@kevintom98 can you be sure nanosleep completed successfully here?

Reply:
Oh yes! That is something i should have considered, the return value of nanosleep(). Thank you for pointing it out Dan! I'll be careful from next time.